UnityProject/Assets/Scripts/Managers/ComponentManager.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using Managers;
+using Messages.Server.SpritesMessages;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class ComponentManager : SingletonManager<ComponentManager>
+{
+	public static Dictionary<GameObject, UniversalObjectPhysics> ObjectToPhysics = new Dictionary<GameObject, UniversalObjectPhysics>();
+	//TODO in future maybe reference a Component that is on every prefab that handles all Components on it
+
+	public override void Awake()
+	{
+		base.Awake();
+
+	}
+
+	public static UniversalObjectPhysics TryGetUniversalObjectPhysics(GameObject gameObject)
+	{
+		if (ObjectToPhysics.ContainsKey(gameObject) == false)
+		{
+			var UOP = gameObject.GetComponent<UniversalObjectPhysics>();
+			if (UOP == null)
+			{
+				Logger.LogError($"Unable to find UniversalObjectPhysics on {gameObject.name}");
+				return null;
+			}
+
+			ObjectToPhysics[gameObject] = UOP;
+		}
+
+		return ObjectToPhysics[gameObject];
+	}
```

Comment:
So as this could be used a lot we can probably do some little optimisations:

- use `ObjectToPhysics.TryGetValue()` for line `21` so we only do a dictionary lookup once, and don't need to do line `33` and `33`.
- In similar fashion as `TryGetValue() let's change this method so that it returns a boolean for if found or not, and uses `out` the component (like the `TryGetComponent()`s), so that usages of this function don't need to check for null again (Unity makes it expensive)